### ISSUE TITLE
Has anyone done any benchmarks on the performance cost to distributed tensorflow caused be networking overheads?

### ISSUE THREAD

**Issue by mattdornfeld** [kubeflow/kubeflow#651]
According to these two papers (http://domino.research.ibm.com/library/cyberdig.nsf/papers/0929052195DD819C85257D2300681E7B/$File/rc25482.pdf, https://arxiv.org/pdf/1711.03386) the only signficant performance cost that comes from using Docker is increased networking latency. Each request seems to have an add overhead of about 100 microseconds. I wonder if this could have any measurable effects on training time for a distributed tensorflow job? Has anyone seen or done these benchmarks?

**Comment by xyhuang** [comment]
Hi, we have tried some simple tests wrt distributed vs local training on docker. To be more specific, we used the cnn-benchmark from tf, and ran some training jobs in the following scenarios in a single workstation, both deployed as docker containers
- local training mode
- distributed training w 1 worker + 1 ps (on the same physical machine)

we did observe that there is significant performance loss w distributed mode in the above scenario, however we have not fully verified if the loss comes all from network overheads or whether there is other reason.

**Comment by mattdornfeld** [comment]
@xyhuang To be clear you noticed a slow down in distributed training when compared to local training?

**Comment by xyhuang** [comment]
@mattdornfeld yes

**Comment by mattdornfeld** [comment]
@xyhuang I think that's too be expected whether or not Docker is used. Even with a single a worker distributed training has to do more operation since gradient updates have to be sent to the parameter server and stored after every batch.
